public/main.js

```diff
@@ -42,103 +36,185 @@ function getTodoList(callback) {
     createRequest.send();
 }
 
+function createButton(id, icon) {
+    var Button = document.createElement("button");
+    Button.setAttribute("type", "button");
+    Button.setAttribute("class", "btn btn-default");
+    Button.setAttribute("id", id);
+    var span = document.createElement("span");
+    span.setAttribute("class", icon);
+    span.setAttribute("aria-hidden", "true");
+    Button.appendChild(span);
+    return Button;
+}
+
+function createReq(method, url, body, errorMsg, callback){
```

Comment:
I just noticed that you are combining the callback and promise from the fetch. While it works, you shouldn't be using callbacks at all, now that you are using promises. `createReq` should return the promise, and the calling code should continue to use that.